apps/server/src/shared/domain/rules/file-record.rule.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@nestjs/common';
+import type { FileRecord, User } from '@shared/domain';
+import { IPermissionContext } from '../interface/permission';
+import { BaseRule } from './base.rule';
+import { CourseRule } from './course.rule';
+
+@Injectable()
+export class FileRecordRule extends BaseRule {
```

Comment:
For the moment the possible generic matches are a lot and not explizit. The rules must be well tested.